feat(amd-gpu): add amd gpu docs page (#21164)
newrelic/docs-website: src/content/docs/infrastructure/host-integrations/host-integrations-list/amd-gpu-integration.mdx

---
title: AMD GPU integration
tags:
  - AMD integration
  - GPU integration
  - New Relic integrations
metaDescription: Use New Relic infrastructure agent to get a dashboard with AMD GPU metrics.
freshnessValidatedDate: 2025-07-08
---

Our AMD GPU integration lets you monitor the status of your AMD GPUs. This integration uses our infrastructure agent with the Flex integration, which lets us access AMD's SMI utility.

<img
  title="AMD GPUs dashboard"
  alt="AMD GPUs dashboard"
  src="/images/infrastructure_screenshot-full_AMD-GPU-dashboard.webp"
/>

<figcaption>

>   After you set up our AMD GPU integration, we give you a dashboard for your GPU metrics.

Reviewer: ```suggestion
  After you set up our AMD GPU integration, you see a dashboard for your GPU metrics.
```

</figcaption>

When you install, you'll get a pre-built dashboard containing crucial GPU metrics:

* GPU utilization
* Memory usage and availability
* Active compute processes
* Clock frequencies
* Temperature monitoring
* Power consumption
* Performance states
* Dynamic and static information about each supported device

<Steps>
  <Step>
    ## Install the infrastructure agent

    To capture data with New Relic, install our infrastructure agent. Our infrastructure agent collects and ingests data so you can keep track of your GPU performance.

    You can install the infrastructure agent two different ways:

    * Our [guided install](https://one.newrelic.com/nr1-core?state=4f81feab-35f7-e97e-9903-52510f8542bd) is a CLI tool that inspects your system and installs the infrastructure agent alongside the application monitoring agent that best works for your system. To learn more about how our guided install works, check out our [Guided install overview](/docs/infrastructure/host-integrations/installation/new-relic-guided-install-overview).

>     * If you'd rather install our infrastructure agent manually, you can follow a tutorial for manual installation for [Linux](/docs/infrastructure/install-infrastructure-agent/linux-installation/install-infrastructure-monitoring-agent-linux), [Windows](/docs/infrastructure/install-infrastructure-agent/windows-installation/install-infrastructure-monitoring-agent-windows/).

Reviewer: ```suggestion
    * If you'd rather install our infrastructure agent manually, follow the tutorial for manual installation for [Linux](/docs/infrastructure/install-infrastructure-agent/linux-installation/install-infrastructure-monitoring-agent-linux), [Windows](/docs/infrastructure/install-infrastructure-agent/windows-installation/install-infrastructure-monitoring-agent-windows/).
```

    FROM AMDGpuSample TIMESERIES AUTO FACET gpu_id
    ```

    ### Power Consumption
    ```sql
    SELECT average(socket_power_w), max(max_power_w) FROM AMDGpuSample TIMESERIES AUTO FACET gpu_id
    ```

    ### Clock Frequencies
    ```sql
    SELECT average(gfx_0_clk_mhz), average(mem_0_clk_mhz), average(fclk_0_clk_mhz) FROM AMDGpuSample TIMESERIES AUTO FACET gpu_id
    ```

    ### GPU Activity by Type
    ```sql
    SELECT average(gfx_activity_percent), average(umc_activity_percent), average(mm_activity_percent) 
    FROM AMDGpuSample TIMESERIES AUTO FACET gpu_id
    ```

    ### ECC Error Monitoring
    ```sql
    SELECT sum(ecc_total_correctable), sum(ecc_total_uncorrectable) FROM AMDGpuSample TIMESERIES AUTO FACET gpu_id
    ```

    ### PCIe Performance
    ```sql
    SELECT average(pcie_bandwidth_mb_s), average(pcie_current_bandwidth_sent), average(pcie_current_bandwidth_received) 
    FROM AMDGpuSample TIMESERIES AUTO FACET gpu_id
    ```

    To create a comprehensive dashboard:

    1. Go to <DNT>**[one.newrelic.com](https://one.newrelic.com/)**</DNT> and click on <DNT>**Dashboards**</DNT>.

>     2. Click on <DNT>**Create a dashboard**</DNT>.

Reviewer: ```suggestion
    2. Click <DNT>**Create a dashboard**</DNT>.
```

    3. Add charts using the NRQL queries provided above.
    4. Customize your dashboard layout and save it.

>     For docs on using and editing dashboards, see [our dashboard docs](/docs/query-your-data/explore-query-data/dashboards/introduction-dashboards).

Reviewer: ```suggestion
    For docs on using and editing dashboards, refer to [our dashboard docs](/docs/query-your-data/explore-query-data/dashboards/introduction-dashboards).
```

### Common issues and solutions:

**AMD SMI not found:**
- Ensure AMD SMI is properly installed and accessible in your PATH
- Verify AMD SMI with: `amd-smi version`

**Permission denied errors:**
- Ensure your user has proper permissions to access GPU devices
- You may need to add your user to the `render` group:
  ```shell
  sudo usermod -a -G render $USER
  ```
- Log out and log back in for changes to take effect

**No GPU detected:**
- Verify AMD GPU is detected by the system: `lspci | grep -i amd`
- Check if amdgpu driver is loaded: `lsmod | grep amdgpu`
- Ensure your AMD GPU is supported by AMD SMI

**Missing metrics:**
- Some older AMD GPUs may not support all sensors
- Check which metrics are available: `amd-smi metric --help`
- Verify GPU compatibility: `amd-smi list`

## What's next? [#next]

You can adapt the Flex configuration to include or exclude information available from the AMD SMI utility. Additional AMD SMI options include:

* `amd-smi list` - List available GPU devices
* `amd-smi static` - Static GPU information
* `amd-smi metric` - Real-time GPU metrics
* `amd-smi process` - Process information using GPU
* `amd-smi monitor` - Continuous monitoring mode

> To learn more about building NRQL queries and generating dashboards, check out these docs:

Reviewer: ```suggestion
To learn more about building NRQL queries and generating dashboards, refer to these docs:
```
